core/corehttp/gateway_handler.go

```diff
@@ -542,7 +537,7 @@ func (i *gatewayHandler) deleteHandler(w http.ResponseWriter, r *http.Request) {
 
 	var newnode *dag.ProtoNode = pbnd
 	for j := len(pathNodes) - 2; j >= 0; j-- {
-		if err := i.node.DAG.Add(ctx, newnode); err != nil {
+		if err := i.node.DAG.Add(tctx, newnode); err != nil {
```

Comment:
Actually, this wasn't a bug. The minute timeout is for _fetching_ the node as that may involve a network operation. Here, we don't really want to time out at all (although the global hour timeout is fine).

Reply:
Ok, got it. I'll make usage of "tctx" more explicit. Just put it into a new method or a "block".